Exiled.Events/EventArgs/ChangingAttachmentsEventArgs.cs

```diff
@@ -7,49 +7,60 @@
-        /// <param name="item"><inheritdoc cref="ChangingAttributesEventArgs.OldItem"/></param>
-        /// <param name="newSight"><inheritdoc cref="NewSight"/></param>
-        /// <param name="newBarrel"><inheritdoc cref="NewBarrel"/></param>
-        /// <param name="newOther"><inheritdoc cref="NewOther"/></param>
-        /// <param name="isAllowed"><inheritdoc cref="ChangingAttributesEventArgs.IsAllowed"/></param>
-        public ChangingAttachmentsEventArgs(Firearm item, SightType newSight, BarrelType newBarrel, OtherType newOther, bool isAllowed = true)
-            : base(item, item, isAllowed)
+        /// <param name="player"><inheritdoc cref="Player"/></param>
+        /// <param name="firearm"><inheritdoc cref="Firearm"/></param>
+        /// <param name="oldIdentifier"><inheritdoc cref="OldAttachmentIdentifier"/></param>
+        /// <param name="newIdentifier"><inheritdoc cref="NewAttachmentIdentifier"/></param>
+        /// <param name="isAllowed"><inheritdoc cref="IsAllowed"/></param>
+        public ChangingAttachmentsEventArgs(
+            Player player,
+            Firearm firearm,
+            AttachmentIdentifier oldIdentifier,
+            AttachmentIdentifier newIdentifier,
+            bool isAllowed = true)
         {
-            NewSight = newSight;
-            NewBarrel = newBarrel;
-            NewOther = newOther;
-            OldItem = (API.Features.Items.Firearm)Item.Get(item);
+            Player = player;
+            Firearm = firearm;
+            OldAttachmentIdentifier = oldIdentifier;
+            NewAttachmentIdentifier = newIdentifier;
+            IsAllowed = isAllowed;
         }
 
-        /// <inheritdoc cref="ChangingAttributesEventArgs.OldItem"/>
-        public new API.Features.Items.Firearm OldItem { get; }
+        /// <summary>
+        /// Gets the <see cref="API.Features.Player"/> who's changing attachments.
+        /// </summary>
+        public Player Player { get; }
+
+        /// <summary>
+        /// Gets the <see cref="API.Features.Items.Firearm"/> which is being modified.
+        /// </summary>
+        public Firearm Firearm { get; }
 
         /// <summary>
-        /// Gets the new item sight attachment.
+        /// Gets the old <see cref="AttachmentIdentifier"/>.
         /// </summary>
-        public SightType NewSight { get; }
+        public AttachmentIdentifier OldAttachmentIdentifier { get; }
 
         /// <summary>
-        /// Gets the new item barrel attachment.
+        /// Gets or sets the new <see cref="AttachmentIdentifier"/>.
         /// </summary>
-        public BarrelType NewBarrel { get; }
+        public AttachmentIdentifier NewAttachmentIdentifier { get; set; }
 
         /// <summary>
-        /// Gets the new item other attachment.
+        /// Gets or sets a value indicating whether or not the attachments can be changed.
         /// </summary>
-        public OtherType NewOther { get; }
+        public bool IsAllowed { get; set; }
     }
 }
```

Comment:
This entire file is a breaking change.

Reply:
Was unimplemented before so I think it's fine

Reply:
@PintTheDragon Yea, you're right, now we can use Attachments properly tho.
It's fine then.